src/hotspot/cpu/x86/x86.ad

```diff
@@ -1938,13 +1967,12 @@ const bool Matcher::match_rule_supported_vector_masked(int opcode, int vlen, Bas
     case Op_AndVMask:
     case Op_OrVMask:
     case Op_XorVMask:
-      assert(bt != T_INT  || VM_Version::supports_avx512bw(), "");
-      assert(bt != T_LONG || VM_Version::supports_avx512bw(), "");
-      if (bt == T_BYTE && !VM_Version::supports_avx512dq()) {
+      if (vlen > 16 && !VM_Version::supports_avx512bw()) {
```

Comment:
Isn't there a limitation as well for vlen 8 and DQ support?

Reply:
That is being taken care in the instruction pattern where we pick masklength to be 16 if target does not support DQ.